cmd/influxd/restore/restore.go

```diff
@@ -197,23 +182,111 @@ func (cmd *Command) unpackMeta(mr *snapshot.MultiReader, sf snapshot.File, confi
 	return nil
 }
 
-func (cmd *Command) unpackData(mr *snapshot.MultiReader, sf snapshot.File, config *Config) error {
-	path := filepath.Join(config.Data.Dir, sf.Name)
-	// Create parent directory for output file.
-	if err := os.MkdirAll(filepath.Dir(path), 0777); err != nil {
-		return fmt.Errorf("mkdir: entry=%s, err=%s", sf.Name, err)
+// unpackShard will look for all backup files in the path matching this shard ID
+// and restore them to the data dir
+func (cmd *Command) unpackShard(shardID string) error {
+	// make sure the shard isn't already there so we don't clobber anything
+	restorePath := filepath.Join(cmd.datadir, cmd.database, cmd.retention, shardID)
+	if _, err := os.Stat(restorePath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("shard already present: %s", restorePath)
+	}
+
+	// find the shard backup files
+	pat := filepath.Join(cmd.backupFilesPath, fmt.Sprintf(backup.BackupFilePattern, cmd.database, cmd.retention, shardID))
+	return cmd.unpackFiles(pat + ".*")
+}
+
+// unpackDatabase will look for all backup files in the path matching this database
+// and restore them to the data dir
+func (cmd *Command) unpackDatabase() error {
+	// make sure the shard isn't already there so we don't clobber anything
+	restorePath := filepath.Join(cmd.datadir, cmd.database)
+	if _, err := os.Stat(restorePath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("database already present: %s", restorePath)
 	}
 
-	// Create output file.
-	f, err := os.Create(path)
+	// find the database backup files
+	pat := filepath.Join(cmd.backupFilesPath, cmd.database)
+	return cmd.unpackFiles(pat + ".*")
+}
+
+// unpackRetention will look for all backup files in the path matching this retention
+// and restore them to the data dir
+func (cmd *Command) unpackRetention() error {
+	// make sure the shard isn't already there so we don't clobber anything
+	restorePath := filepath.Join(cmd.datadir, cmd.database, cmd.retention)
+	if _, err := os.Stat(restorePath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("retention already present: %s", restorePath)
+	}
+
+	// find the retention backup files
+	pat := filepath.Join(cmd.backupFilesPath, cmd.database)
+	return cmd.unpackFiles(fmt.Sprintf("%s.%s.*", pat, cmd.retention))
+}
+
+// unpackFiles will look for backup files matching the pattern and restore them to the data dir
+func (cmd *Command) unpackFiles(pat string) error {
+	fmt.Printf("restoring from backup %s\n", pat)
```

Comment:
Trivial, but since this is a statement to `stdout` (as opposed to a log message or error), the first letter would look better capitalized.